fido/src/main/java/com/yubico/yubikit/fido/ctap/Hkdf.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.fido.ctap;
+
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import javax.annotation.Nullable;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implements HMAC-based Extract-and-Expand Key Derivation Function (HKDF)
+ *
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc5869">rfc5869</a>
+ */
+public class Hkdf {
+
+    private final int length;
+    private final byte[] salt;
+    private final byte[] info;
+    private final Mac mac;
+
+    public Hkdf(String algo, @Nullable byte[] salt, byte[] info, int length) throws NoSuchAlgorithmException {
```

Comment:
I find it a little hard to follow where some of these methods use arguments that shadow the instance members. For example, `extract` takes `salt` even though it could use `this.salt`. On the other hand, `expand` uses `(this.)length` and `(this.)info`.
I'm also not sure why the constructor accepts a null `salt`, as we always explicitly pass it.

Reply:
Updated in [18961b5](https://github.com/Yubico/yubikit-android/pull/105/commits/18961b55f93a80af3f8a02221a993b3567aaa687) 

`extract` and `expand` now also have the same parameters as in the rfc. See [2.2 Step 1: Extract](https://datatracker.ietf.org/doc/html/rfc5869#section-2.2) and [2.3 Step2: Expand](https://datatracker.ietf.org/doc/html/rfc5869#section-2.3)